numpy/random/tests/test_extending.py

```diff
@@ -57,9 +58,9 @@ def test_cython(tmp_path):
     # gh-16162: make sure numpy's __init__.pxd was used for cython
     # not really part of this test, but it is a convenient place to check
     with open(build_dir / 'extending.c') as fid:
-        txt_to_find = 'NumPy API declarations from "numpy/__init__.pxd"'
+        txt_to_find = re.compile(r'NumPy API declarations from "numpy\/__init__(\.([^.]*?))?.pxd"')
         for i, line in enumerate(fid):
-            if txt_to_find in line:
+            if txt_to_find.search(line):
                 break
```

Comment:
This looks like a bugfix for the new cython change, right? Perhaps we should create a new PR with this.